Gridy/Controller/ImageEditorViewController.swift

```diff
@@ -154,46 +159,81 @@ class ImageEditorViewController: UIViewController, UIGestureRecognizerDelegate {
-                }) { (success) in }
+                        self.chosenImageView.transform = .identity
+                    }) { (success) in }
             }
         default:
             break
         }
-            
-            sender.scale = 1
+        
+        sender.scale = 1
+    }
+    
+    func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer,
+                           shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer)
+    -> Bool {
+        // simultaneous gesture recognition will only be supported for chosenImageView
+        if gestureRecognizer.view != chosenImageView {
+            return false
         }
         
-        func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer,
-                               shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer)
-            -> Bool {
-                // simultaneous gesture recognition will only be supported for chosenImageView
-                if gestureRecognizer.view != chosenImageView {
-                    return false
-                }
-                
-                // neither of the recognized gestures should not be tap gesture
-                if gestureRecognizer is UITapGestureRecognizer
-                    || otherGestureRecognizer is UITapGestureRecognizer
-                    || gestureRecognizer is UIPanGestureRecognizer
-                    || otherGestureRecognizer is UIPanGestureRecognizer {
-                    return false
-                }
-                
-                return true
+        // neither of the recognized gestures should not be tap gesture
+        if gestureRecognizer is UITapGestureRecognizer
+            || otherGestureRecognizer is UITapGestureRecognizer
+            || gestureRecognizer is UIPanGestureRecognizer
+            || otherGestureRecognizer is UIPanGestureRecognizer {
+            return false
         }
-
-    func cropImage(screenshot: UIImage) -> UIImage {
-        let crop = self.chosenImageView.convert(self.gridFrameView.bounds, from: self.gridFrameView)
-        let cgImage = screenshot.cgImage!.cropping (to: crop)
-        let image: UIImage = UIImage (cgImage: cgImage!)
-        print(crop)
-        return image
+        
+        return true
+    }
+    
+    private func crop(image: UIImage) -> UIImage {
+        let scale: CGFloat = image.scale
+        let rectangle = self.chosenImageView.convert(self.gridFrameView.bounds, from: self.gridFrameView)
+        let scaledRectangle = CGRect(x: rectangle.origin.x * scale,
+                                     y: rectangle.origin.y * scale,
+                                     width: rectangle.size.width * scale,
+                                     height: rectangle.size.height * scale)
+        let cgImage = image.cgImage!.cropping (to: scaledRectangle)
+        let croppedImage = UIImage(cgImage: cgImage!, scale: scale, orientation: .up)
+        return croppedImage
+    }
```

Comment:
I look at your crop bug, the good news is that your crop function is working well :)
You have a bug because you apply the crop function on a different image than the one you have shown on screen. Look well at the image on screen versus the one in Assets/RandomImages, and you will see that they are quite not the same. I see different approaches to fix that issue, but I am curious to see how you will tackle it :) Send me a message if you are blocked.